### PR TITLE
fix: function expand in tensor.py

### PR DESCRIPTION
**Pull request by WannaFy**
According to Pytorch, we should delete the extra dimensions in the beginning with size 1 to make the shape of the gradient exactly equal that of the input.

### CODE REVIEW

**Comment by srush** [comment]
Can you make this fix a bit more clear? What was the error?

**Comment by WannaFy** [comment]
I am so sorry for not explaining this fix clearly last night(I was so sleepy because it was too late in China).
Please run the following code
```python

import minitorch
from minitorch.tensor_functions import tensor as tensor_from_list

a=tensor_from_list([1],requires_grad=True)
b=tensor_from_list([[1,1]],requires_grad=True)

c=(a*b).sum()

c.backward()
print(c.shape)
print(a.shape,a.grad.shape)
print(b.shape,b.grad.shape)
```
In the original version, the console will print
```shell
(1,)
(1,) (1, 1)
(1, 2) (1, 2)
```
in which a.shape!=a.grad.shape.

While in my version and also in pytorch, a.shape=a.grad.shape.
```shell
(1,)
(1,) (1,)
(1, 2) (1, 2)
```
Again, thanks for your great tutorial.

**Comment by WannaFy** [comment]
I just checked out the module-3 which is still in the old version,  the function 'expand' does  make the shape of the gradient exactly equal that of the input.

**Comment by WannaFy** [comment]
I am also wondering  whether copying files from the new version of module-2 directly to the old version of module-3 will affect the later implementation.  Thank you so much for your reply.

**Comment by WannaFy** [comment]
```python
out.shape=self.shape
```

**Comment by srush** [comment]
Oh I see. thanks! I will add your test to the main library and see if I can find a fix.

**Comment by srush** [comment]
Added in https://github.com/minitorch/Module-2/pull/15 (this code is automatically generated from the template repo.)
